src/programs.rkt

```diff
@@ -148,12 +143,13 @@
          (define atypes
            (match (operator-info op 'itype)
              [(? list? as) as]
-             [(? type-name? a) (map (const a) args)]))
+             [(? symbol? a) (map (const a) args)]))
          (unless (= (length atypes) (length args))
            (raise-argument-error 'eval-prog "expr?" prog))
          (cons (operator-info op mode)
-               (for/list ([arg args] [atype atypes])
-                 (munge arg (get-representation* atype))))]
+               (for/list ([arg args] [type atypes])
+                (let ([type* (if (equal? type 'real) 'binary64 type)]) ; TODO: temp conversion
+                 (munge arg (get-representation type*)))))]
```

Comment:
Is this still necessary? I think all of the `atypes` are now representation names.